test/e2e/helpers.go

```diff
@@ -823,3 +829,94 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 	}
 	return b.String()
 }
+
+func WaitForClusterKubeconfigSecret(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
```

Comment:
this might be a helper in the Cluster API test framework already. If not, it should be. Nothing CAPZ specific here.

Reply:
Good call-out. In fact we don't need to run these kubeconfig and running nodes checks, as the retryable "install helm chart" func will absorb the failures due to the workload cluster not being actually running yet.